protocol/contracts/beanstalk/field/FieldFacet.sol

```diff
@@ -167,89 +169,102 @@ contract FieldFacet is Invariable, ReentrancyGuard {
      */
     function _harvestPlot(
         address account,
+        uint256 fieldId,
         uint256 index
     ) private returns (uint256 harvestablePods) {
         // Check that `account` holds this Plot.
-        uint256 pods = s.accounts[account].field.plots[index];
+        uint256 pods = s.accounts[account].fields[fieldId].plots[index];
         require(pods > 0, "Field: no plot");
 
         // Calculate how many Pods are harvestable.
         // The upstream _harvest function checks that at least some Pods
         // are harvestable.
-        harvestablePods = s.field.harvestable.sub(index);
-        delete s.accounts[account].field.plots[index];
-
-        // Cancel any active Pod Listings active for this Plot.
-        // Note: duplicate of {Listing._cancelPodListing} without the
-        // ownership check, which is done above.
-        if (s.podListings[index] > 0) {
-            delete s.podListings[index];
-            emit PodListingCancelled(LibTractor._user(), index);
-        }
+        harvestablePods = s.fields[fieldId].harvestable.sub(index);
+
+        // TODO: Verify this works. Solicit review.
+        // _cancelPodListing(LibTractor._user(), fieldId, index);
+        address(this).delegatecall(
+            abi.encodeWithSelector(IBeanstalk.cancelPodListing.selector, fieldId, index)
+        );
```

Comment:
@Brean0 

this is a call from one facet to a different facet's public function. i am unsure about the implications of this design. i could not find any other examples in the codebase of similar behavior. wdyt about this approach?

Reply:
lgtm - ideally `_cancelPodListing` would've been a function from a library but given it was written prior, in this context it makes sense to me.

Reply:
moved it into a lib here
https://github.com/BeanstalkFarms/Beanstalk/pull/860/commits/0575d331733ccf2c10cdf48c0140458a3b8b922d#diff-d1df16bc4f6f3b8f37c60a5dab5717e751f8a2986236fc936965d2ecd0f0bdf6R198

i am not sure what blocked me from doing that in the first place, but agree this is the better solution.